### DISCUSSION TITLE
Simple sobol example updated for FLORIS v3?

### DISCUSSION THREAD

**Discussion by mr483** [NREL/flasc#9] [Q&A] [answered]
Hi Bart, 

I am hoping to do some FLORIS model tuning, and came across the FLASC package which has lots of useful features for my purposes - thanks very much for releasing this! 

I'm particularly hoping to use the floris_sensitivity_analysis module, but it seems perhaps it hasn't yet been updated to be compatible for FLORIS 3.0. I'm debugging your simple_sobol_example.py and updating the syntax to be compatible with the new FLORIS. 

However, I've now run into an issue with floris_tools.calc_floris(), as I am receiving the error that use_model_params is not yet implemented in FLORIS 3.0. I assume the problem is providing sets of model parameters rather than fixed values to FLORIS. Is there any easy fix I could implement to get the sobol analysis working? 

One more question: It appears to me that currently floris_sensitivity_analysis.calculate_wfpower_for_samples() uses a single set of wind conditions for all of the sobol samples. Is this correct? Is there a way to extend this to sets of conditions and perhaps use a summary statistic such as AEP across the conditions as the outputs? Perhaps this is a different application than what you intended in this module.

**Comment by Bartdoekemeijer** [comment, marked as answer]
Hi @mr483,

Great to hear you are finding FLASC useful! The sensitivity analysis module has indeed not yet been updated. Essentially, I still have to reintroduce a functionality in `calc_floris` to overwrite model parameters. The easiest way to get this working is by using the suggestions in [this post on the floris boards](https://github.com/NREL/floris/discussions/399). It is likely going to be something like Raf's suggestion:

```
d_floris = fi.floris.as_dict()
d_floris["wake"]["model_strings"]["velocity_model"] = "cc"
d_floris["wake"]["wake_velocity_parameters"]["cc"] = {
    "a_s": 0.179367259,
    "b_s": 0.0118889215,
    "c_s1": 0.0563691592,
    "c_s2": 0.13290157,
    "a_f": 3.11,
    "b_f": -0.68,
    "c_f": 2.41,
    "alpha_mod": 1.0,
}
fi = FlorisInterface(d_floris)
```

where we then assign the values `a_s`, `b_s`, and so on, based on the values we specify in the Pandas dataframe in `calc_floris`.

Also, concerning your last question, indeed the current example only uses a single floris condition to calculate the sensitivity index. You can easily extend this to AEP. Essentially, the choice of which conditions to include in your sensitivity cost function depends on the application you envision. If you are tuning model parameters based on specific SOWFA simulations, you can model those SOWFA simulations in FLORIS and see which parameters have the highest Sobol index (i.e., have the largest impact on the output you are trying to tune your model to). If you, generally, are looking at which parameters have the largest impact on the AEP, I think what you describes makes more sense.

Best of luck! Do let me know if you run into any other issues. Also, if you manage to re-introduce the `use_model_params` functionality, please consider opening a pull request so that I can get this merged in!

Thanks!

Best,
Bart
